test/Microsoft.IdentityModel.Tokens.Tests/ValidatorsTests.cs

```diff
@@ -37,40 +37,258 @@ namespace Microsoft.IdentityModel.Tokens.Tests
+                    },
+                    new AudienceValidationTheoryData
+                    {
+                        Audiences = new List<string> { "audience1" },
+                        ExpectedException = ExpectedException.SecurityTokenInvalidAudienceException("IDX10208:"),
+                        TestId = "ValidateAudienceTrueValidAudienceAndValidAudiencesNull"
+                    }
+                };
+            }
+        }
 
-                var dataset = new TheoryData<List<string>, SecurityToken, TokenValidationParameters, ExpectedException>();
+        [Theory, MemberData(nameof(ValidateAudienceTheoryData))]
+        public void ValidateAudience(AudienceValidationTheoryData theoryData)
+        {
+            var context = TestUtilities.WriteHeader($"{this}.ValidateAudience", theoryData);
+            try
+            {
+                Validators.ValidateAudience(theoryData.Audiences, theoryData.SecurityToken, theoryData.TokenValidationParameters);
+                theoryData.ExpectedException.ProcessNoException(context);
+            }
+            catch (Exception ex)
+            {
+                theoryData.ExpectedException.ProcessException(ex, context);
+            }
 
-                dataset.Add(null, null, null, ExpectedException.ArgumentNullException());
-                dataset.Add(null, null, new TokenValidationParameters { ValidateAudience = false }, ExpectedException.NoExceptionExpected);
-                dataset.Add(null, null, new TokenValidationParameters(), ExpectedException.SecurityTokenInvalidAudienceException("IDX10207:"));
-                dataset.Add(audiences, null, new TokenValidationParameters(), ExpectedException.SecurityTokenInvalidAudienceException("IDX10208:", propertiesExpected: properties));
-                dataset.Add(audiences, null, new TokenValidationParameters { ValidAudience = NotDefault.Audience }, ExpectedException.SecurityTokenInvalidAudienceException("IDX10214:", propertiesExpected: properties));
-                dataset.Add(audiences, null, new TokenValidationParameters { ValidAudiences = invalidAudiences }, ExpectedException.SecurityTokenInvalidAudienceException("IDX10214:", propertiesExpected: properties));
-                dataset.Add(audiences, null, new TokenValidationParameters { ValidAudience = Default.Audience }, ExpectedException.NoExceptionExpected);
-                dataset.Add(audiences, null, new TokenValidationParameters { ValidAudiences = audiences }, ExpectedException.NoExceptionExpected);
+            TestUtilities.AssertFailIfErrors(context);
+        }
 
-                return dataset;
+        public static TheoryData<AudienceValidationTheoryData> ValidateAudienceTheoryData
+        {
+            get
+            {
+                var audience1 = "http://audience1.com";
+                var audience2 = "http://audience2.com";
+                List<string> audiences1 = new List<string> { "", audience1 };
```

Comment:
Are the empty strings added to these list so that the code below is tested? https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/5315c1404d94c596ea92bf267411b4e05ee515af/src/Microsoft.IdentityModel.Tokens/Validators.cs#L92-L96

Reply:
yes